interfaces.go

```diff
@@ -15,15 +15,25 @@ type Connection interface {
 
 // KeySpace is used to obtain tables from.
 type KeySpace interface {
-	MapTable(tableName, id string, row interface{}) MapTable
-	MultimapTable(tableName, fieldToIndexBy, uniqueKey string, row interface{}) MultimapTable
-	MultimapMultiKeyTable(tableName string, fieldToIndexBy, uniqueKey []string, row interface{}) MultimapMkTable
-	TimeSeriesTable(tableName, timeField, uniqueKey string, bucketSize time.Duration, row interface{}) TimeSeriesTable
-	MultiTimeSeriesTable(tableName, fieldToIndexByField, timeField, uniqueKey string, bucketSize time.Duration, row interface{}) MultiTimeSeriesTable
-	MultiKeyTimeSeriesTable(tableName string, fieldToIndexByField []string, timeField string, uniqueKey []string, bucketSize time.Duration, row interface{}) MultiKeyTimeSeriesTable
-	FlakeSeriesTable(tableName, idField string, bucketSize time.Duration, row interface{}) FlakeSeriesTable
-	MultiFlakeSeriesTable(tableName, indexField, idField string, bucketSize time.Duration, row interface{}) MultiFlakeSeriesTable
-	Table(tableName string, row interface{}, keys Keys) Table
+	// MapTable is a simple key-value store.
+	MapTable(prefixForTableName, partitionKey string, row interface{}) MapTable
+	// MultimapTable lets you list rows based on a field equality, eg. 'list all sales where seller id = v'.
+	// It uses the partitionKey for partitioning the data. The ordering within a partition is determined using the clusteringKey.
+	MultimapTable(prefixForTableName, partitionKey, clusteringKey string, row interface{}) MultimapTable
+	// // MultimapMkTable lets you list rows based on several fields equality, eg. 'list all sales where seller id = v and name = 'john'.
+	// It uses the partitionKey for partitioning the data. The ordering within a partition is determined by a set of clusteringKeys.
+	MultimapMultiKeyTable(prefixForTableName string, partitionKey, clusteringKey []string, row interface{}) MultimapMkTable
+	/*
+		TimeSeriesTable lets you list rows which have a field value between two date ranges.
+		timeField is used as the partition key
+		bucketSize is used to determine for what duration the data will be stored on the same partition.
+	*/
+	TimeSeriesTable(prefixForTableName, timeField, clusteringKey string, bucketSize time.Duration, row interface{}) TimeSeriesTable
+	MultiTimeSeriesTable(prefixForTableName, partitionKey, timeField, clusteringKey string, bucketSize time.Duration, row interface{}) MultiTimeSeriesTable
+	MultiKeyTimeSeriesTable(prefixForTableName string, partitionKeys []string, timeField string, clusteringKeys []string, bucketSize time.Duration, row interface{}) MultiKeyTimeSeriesTable
+	FlakeSeriesTable(prefixForTableName, idField string, bucketSize time.Duration, row interface{}) FlakeSeriesTable
+	MultiFlakeSeriesTable(prefixForTableName, indexField, idField string, bucketSize time.Duration, row interface{}) MultiFlakeSeriesTable
```

Comment:
Likewise, could we improve `indexField` and `idField`?

Reply:
Just noticed that even though `MultiTimeSeriesTable` has a clustering key, `MultiFlakeSeriesTable` doesn't. I wonder why that's the case because the similar naming suggests that they behave the same way, one being a time series and the other using a flakeId.